bin/download_caddy.bash

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bash
+
+set -eu
+
+# The goal here is to
+# Check releases and their checksums, and update the associated data here.
+# @link https://github.com/caddyserver/caddy/releases
+caddy_version="2.4.3"
+
+function output {
+    style_start=""
+    style_end=""
+    if [ "${2:-}" != "" ]; then
+    case $2 in
+        "success")
+            style_start="\033[0;32m"
+            style_end="\033[0m"
+            ;;
+        "error")
+            style_start="\033[31;31m"
+            style_end="\033[0m"
+            ;;
+        "info"|"warning")
+            style_start="\033[33m"
+            style_end="\033[39m"
+            ;;
+        "heading")
+            style_start="\033[1;33m"
+            style_end="\033[22;39m"
+            ;;
+    esac
+    fi
+
+    builtin echo -e "${style_start}${1}${style_end}"
+}
+
+
+kernel=$(uname -s 2>/dev/null || /usr/bin/uname -s)
+case ${kernel} in
+    "Linux"|"linux")
+        kernel="linux"
+        ;;
+    "Darwin"|"darwin")
+        kernel="mac"
+        ;;
+    *)
+        output "Your OS '${kernel}' not supported" "error"
+        exit 1
+        ;;
+esac
+
+
+machine=$(uname -m 2>/dev/null || /usr/bin/uname -m)
+case ${machine} in
+    arm|armv7*)
+        machine="arm"
+        ;;
+    aarch64*|armv8*|arm64)
+        machine="arm64"
+        ;;
+    i[36]86)
+        machine="386"
+        ;;
+    x86_64)
+        machine="amd64"
+        ;;
+    *)
+        output "Your architecture '${machine}' is not currently supported" "error"
+        exit 1
+        ;;
+esac
+
+platform="${kernel}_${machine}"
+
+output "Platform: ${platform}" "info"
+
+checksums=$(curl -sSL "https://github.com/caddyserver/caddy/releases/download/v${caddy_version}/caddy_${caddy_version}_checksums.txt")
```

Comment:
This kinda defeats the purpose btw, because if someone was able to man-in-the-middle your script, they could also just replace the checksums file with the ones that match their malicious binaries. You should manually download the checksums and insert them in your script whenever you bump the version, or just skip the checksum validation step altogether.

Reply:
I will remove checksums validation. It's something that is executed at build-time only anyway, and shouldn't have any impact for the end-user